internal/api/token.go

```diff
@@ -577,6 +589,95 @@ func (a *API) IdTokenGrant(ctx context.Context, w http.ResponseWriter, r *http.R
 	return sendJSON(w, http.StatusOK, token)
 }
 
+func (a *API) OAuthPKCE(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	db := a.db.WithContext(ctx)
+	var grantParams models.GrantParams
+
+	params := &PKCEGrantParams{}
+	body, err := getBodyBytes(r)
+	if err != nil {
+		return internalServerError("Could not read body").WithInternalError(err)
+	}
+
+	if err = json.Unmarshal(body, params); err != nil {
+		return badRequestError("invalid body: unable to parse JSON").WithInternalError(err)
+	}
+
+	if params.AuthCode == "" || params.CodeVerifier == "" {
+		return badRequestError("invalid request: both auth code and code verifier should be non-empty")
+	}
+
+	flowState, err := models.FindFlowStateByAuthCode(db, params.AuthCode)
+	// Sanity check in case user ID was not set properly
+	if models.IsNotFoundError(err) || flowState.UserID == uuid.Nil {
```

Comment:
Is `UserID` nullable? Can we try to use `*uuid.UUID` for the null case so we prevent issues like we had in the past?